### PR TITLE
Post Comments block - add style supports for text and background settings.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
## Description

Adds some style support for the Post Comments block:
* Line Height
* Font Size
* Font and Background colors
* Text Alignment

As well as fixes an issue where links in the post comments were not appearing as links in the editor:
Screen Shot 2020-07-21 at 2 19 30 PM

Note - there may be a handful of issues between editor and front-end parity for this block and some future work to be done ahead as noted in https://github.com/WordPress/gutenberg/issues/24101

## How has this been tested?

Verify these settings persist across the editor and front-end.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/post-comments/edit.js [hunk 1]
```diff
@@ -18,19 +28,56 @@ function PostCommentsDisplay( { postId } ) {
 			// TODO: "No Comments" placeholder should be editable.
 			return comments && comments.length
 				? comments.map( ( comment ) => (
-						<p key={ comment.id }>{ comment.content.raw }</p>
+						<p key={ comment.id }>
+							<RawHTML>{ comment.content.raw }</RawHTML>
+						</p>
```
Comment (Copons) [inline packages/block-library/src/post-comments/edit.js:33]: Wrapping `RawHTML` inside a `p` is semantically incorrect because `RawHTML` renders as a `div`.

I think this could be improved in several ways:

1. Use the comment content's `rendered` property instead of `raw`.
`comment.content.raw` is what the user submitted (or, more precisely, what has been saved in the DB after sanitization), and it's supposed to be used when editing the comment.
`comment.content.rendered` is what WP uses when displaying the comment.
In this case, we are not editing but displaying, so `rendered` is more appropriate.
It also automatically wraps paragraphs (as in: blocks of text separated by 2 newlines) in `p` elements! ✨ 

2. Pass a CSS class to the mapped comment (something like `.wp-block-post-comments__comment`), which should make it easier to style it.
Not strictly needed in this PR, but eventually we will need to visually separate comments between each other!

3. Drop the `p` and just return a map of `RawHTML`.


Eventually it would look somewhat like this:
```jsx
return comments.map( comment =>
  <RawHTML className="wp-block-post-comments__comment" key={ comment.id }>
    { comment.content.rendered }
  </RawHTML>
);
```

File: packages/block-library/src/post-comments/edit.js [hunk 2]
```diff
@@ -18,19 +28,56 @@ function PostCommentsDisplay( { postId } ) {
...
+
+	if ( ! postType || ! postId ) {
+		return (
+			<Warning>{ __( 'Post comments block: no post found.' ) }</Warning>
+		);
+	} else if ( postType !== 'post' ) {
+		return (
+			<Warning>
+				{ __(
+					'Post comments block: Comments are only available in posts. Please add this block to a post instead.'
```
Comment (Copons) [inline packages/block-library/src/post-comments/edit.js:57]: I don't really agree with this, as it's not strictly true.

I'd rather we checked if the post type (or even the post!) supports comments, than just relying on this check.

File: packages/block-library/src/post-comments/edit.js (hunk 1 above)
Comment (Addison-Stavlo) [reply packages/block-library/src/post-comments/edit.js:33]: > Wrapping RawHTML inside a p is semantically incorrect because RawHTML renders as a div.

I was under the impression this was not the case:
https://github.com/WordPress/gutenberg/blob/24c6114df4ff3fd18c34fa258a42f1130577f915/packages/element/src/raw-html.js#L23-L24

Comment (Addison-Stavlo) [reply packages/block-library/src/post-comments/edit.js:33]: > Use the comment content's rendered property instead of raw.

I do think this makes sense.  I will have to test it out.  I wonder why it was using `raw` in the first place.

Comment (Copons) [reply packages/block-library/src/post-comments/edit.js:33]: > I was under the impression this was not the case:

Yeah, I kinda think that comment is not exactly correct. 😄 
Either way, we would be passing more props (the `key`, possibly a `className`), so the `div` should better show up!
_(it does in my tests)_

Comment (Addison-Stavlo) [reply packages/block-library/src/post-comments/edit.js]: Awesome, these suggestions make things work much better. Thanks!  
The rendered content also has `<br/>`s in it as well which makes it look much better and more in-line with what is on the front end.

File: packages/block-library/src/post-comments/edit.js (hunk 2 above)
Comment (Addison-Stavlo) [reply packages/block-library/src/post-comments/edit.js]: I removed this extra check for now, so currently it will just fall back to 'No comments.' in that odd case.

> I'd rather we checked if the post type (or even the post!) supports comments

That sounds like a good way to do this!  Although I'm not sure how to go about checking that currently and it may be best to wait as it seems like this block will require some significant restructuring in the future (#24101).

Comment (Copons) [reply packages/block-library/src/post-comments/edit.js]: Oh yes definitely!
Right now I see this block as almost like a placeholder for the comments loop, so I think it's reasonable to keep chiseling it into shape.

The feature support check should be a major concern for when we move to the multipage editing, especially considering how extensible _everything_ in WP is.
Anything could be added or removed from any post types, or even posts (e.g. comments can be disabled on specific posts with the Allow Comments toggle in the document sidebar).
But, this affects almost all FSE blocks, so we'll need to figure out a proper solution that fits most/all use cases.

tl;dr this PR is not the most appropriate place to figure it out. 🙂

**Review by Copons — APPROVED** [review]
